pandas/core/dtypes/dtypes.py

```diff
@@ -760,10 +786,15 @@ def __new__(cls, freq=None):
             return cls._cache[freq.freqstr]
         except KeyError:
             u = object.__new__(cls)
-            u.freq = freq
+            u._freq = freq
             cls._cache[freq.freqstr] = u
             return u
 
+    @property
```

Comment:
you changed this mainly to have a proper doc-string?

Reply:
Initially yes, but in the end it also fixes https://github.com/pandas-dev/pandas/issues/26096 (ensure `freq` is not writable by the user), so I think it is a good idea in general (it's also what CategoricalDtype does with categories/codes and DatetimeTZDtype with unit/tz)

Reply:
yes this is good (prob could add a @property.setter to get a nice error message but can be followup)